test/e2e/common/loki/loki.go

```diff
@@ -0,0 +1,84 @@
+package loki
+
+import (
+	"bytes"
+	"e2e-basic/common/e2e_config"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"os"
+	"strconv"
+	"sync"
+	"time"
+
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+var g_apiUser string
+var g_apiPw string
+var g_buildNumber string
+var g_once sync.Once
+
+func SendLokiMarker(text string) error {
+	g_once.Do(func() {
+		g_apiUser = os.Getenv("grafana_api_user")
```

Comment:
I'd say move retrieval of the image tag inside the Once function, then make the decision whether adding markers is possible in that function and store as boolean.
Next time round just check the boolean
Also report errors from within the Once function? 
Not sure about returning an error, caller should  either log the error message or ignore but proceed.
IMO not much point in seeing the same error message multiple times.

Reply:
> I'd say move retrieval of the image tag inside the Once function
But then the image tag would have to be stored as another static variable, mirroring the one inside the config struct which is itself cached. Not sure if caching the same value twice makes sense.
The failure is going to be a configuration issue which will happen rarely. I'm not sure if saving a few extra entries in the log is worth the extra complexity of caching the failure.

Reply:
OTOH we could end up with a lot of errors when the loki marker function is called a lot. 
Also could save some processing when run outside of CI with loki not enabled.
I'll make the change for the code to decide once whether loki is enabled.